test/templates/template_test.rb

```diff
@@ -17,15 +17,25 @@ def setup
 
   def test_template_requests_return_data_and_contain_template_url
     @expected_body = "my template"
+    @args = { :token => "test-token" }
     stub_request(:get, @foreman_url+'/unattended/provision?token=test-token&url='+@template_url).to_return(:status => [200, 'OK'], :body => @expected_body)
-    result = Proxy::Templates::Handler.get_template('provision', 'test-token')
+    result = Proxy::Templates::Handler.get_template('provision', @args)
     assert_equal(@expected_body, result)
   end
 
   def test_template_requests_pass_static_flag_to_foreman
     @expected_body = "my template"
+    @args = { :token => "test-token" }
     stub_request(:get, @foreman_url+'/unattended/provision?static=true&token=test-token&url='+@template_url).to_return(:status => [200, 'OK'], :body => @expected_body)
-    result = Proxy::Templates::Handler.get_template('provision', 'test-token', 'true')
+    result = Proxy::Templates::Handler.get_template('provision', @args, 'true')
+    assert_equal(@expected_body, result)
+  end
+
+  def test_template_requests_with_macaddress
+    @expected_body = "my template"
+    @args = { :mac => "aa:bb:cc:dd:ee:ff" }
+    stub_request(:get, @foreman_url+'/unattended/provision?static=true&mac=aa:bb:cc:dd:ee:ff&url='+@template_url).to_return(:status => [200, 'OK'], :body => @expected_body)
+    result = Proxy::Templates::Handler.get_template('provision', @args)
```

Comment:
expected url has ```static=true```, but when you are making the call, ```static``` parameter is set to default value of ```false```